File: src/api/services/party.ts
import request from '../request'
import type { ApiResponse, PageResult } from '@/types'

export interface PartyGroup {
  id: string
  name: string
  parentId: string | null
  leaderName: string
  memberCount: number
  order: number
  createdAt: string
}

export interface PartyMember {
  id: string
  name: string
  groupId: string
  groupName: string
  position: string
  joinDate: string
  phone: string
  status: 'active' | 'transferred' | 'inactive'
}

export interface TransferRecord {
  id: string
  memberId: string
  memberName: string
  fromGroupId: string
  fromGroupName: string
  toGroupId: string
  toGroupName: string
  reason: string
  transferDate: string
}

export function getGroupTree(): Promise<ApiResponse<PartyGroup[]>> {
  return request.get('/party/groups/tree')
}

export function createGroup(data: Partial<PartyGroup>): Promise<ApiResponse<PartyGroup>> {
  return request.post('/party/groups', data)
}

export function updateGroup(id: string, data: Partial<PartyGroup>): Promise<ApiResponse<PartyGroup>> {
  return request.put(`/party/groups/${id}`, data)
}

export function deleteGroup(id: string): Promise<ApiResponse<null>> {
  return request.delete(`/party/groups/${id}`)
}

export function getMembers(params: {
  page: number
  pageSize: number
  keyword?: string
  groupId?: string
  status?: string
}): Promise<ApiResponse<PageResult<PartyMember>>> {
  return request.get('/party/members', { params })
}

export function getAllMembers(): Promise<ApiResponse<PartyMember[]>> {
  return request.get('/party/members/all')
}

export function createMember(data: Partial<PartyMember>): Promise<ApiResponse<PartyMember>> {
  return request.post('/party/members', data)
}

export function updateMember(id: string, data: Partial<PartyMember>): Promise<ApiResponse<PartyMember>> {
  return request.put(`/party/members/${id}`, data)
}

export function deleteMember(id: string): Promise<ApiResponse<null>> {
  return request.delete(`/party/members/${id}`)
}

export function getTransfers(params: {
  page: number
  pageSize: number
  keyword?: string
}): Promise<ApiResponse<PageResult<TransferRecord>>> {
  return request.get('/party/transfers', { params })
}

export function transferMember(data: {
  memberId: string
  toGroupId: string
  reason: string
}): Promise<ApiResponse<TransferRecord>> {
  return request.post('/party/transfers', data)
}

File: backend/models/party.py
from typing import Optional
from pydantic import BaseModel


class PartyGroup(BaseModel):
    id: str
    name: str
    parentId: Optional[str] = None
    leaderName: str
    memberCount: int
    order: int
    createdAt: str


class GroupCreate(BaseModel):
    name: str
    parentId: Optional[str] = None
    leaderName: str = ""
    order: int = 0


class GroupUpdate(BaseModel):
    name: Optional[str] = None
    parentId: Optional[str] = None
    leaderName: Optional[str] = None
    order: Optional[int] = None


class PartyMember(BaseModel):
    id: str
    name: str
    groupId: str
    groupName: str
    position: str
    joinDate: str
    phone: str
    status: str


class MemberCreate(BaseModel):
    name: str
    groupId: str
    groupName: str = ""
    position: str = "组员"
    joinDate: str = ""
    phone: str = ""
    status: str = "active"


class MemberUpdate(BaseModel):
    name: Optional[str] = None
    groupId: Optional[str] = None
    groupName: Optional[str] = None
    position: Optional[str] = None
    joinDate: Optional[str] = None
    phone: Optional[str] = None
    status: Optional[str] = None


class TransferRecord(BaseModel):
    id: str
    memberId: str
    memberName: str
    fromGroupId: str
    fromGroupName: str
    toGroupId: str
    toGroupName: str
    reason: str
    transferDate: str


class TransferCreate(BaseModel):
    memberId: str
    toGroupId: str
    reason: str
